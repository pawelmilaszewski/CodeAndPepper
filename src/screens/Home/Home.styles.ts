import {StyleSheet} from 'react-native';
import {Colors} from '../../utils/colors';

export const styles = StyleSheet.create({
  main: {
    flex: 1,
    paddingTop: 54,
    paddingHorizontal: 20,
  },
  text: {
    fontSize: 24,
    letterSpacing: 1,
    lineHeight: 26,
    marginTop: 32,
    marginBottom: 16,
  },
  title: {
    fontSize: 32,
    letterSpacing: 1,
    lineHeight: 36,
    textAlign: 'center',
  },
  playButton: {
    backgroundColor: Colors.GREEN,
    borderRadius: 10,
    padding: 16,
    width: '80%',
    alignItems: 'center',
    alignSelf: 'center',
    justifyContent: 'center',
    marginTop: 64,
  },
  playButtonText: {
    fontSize: 24,
    fontWeight: 'bold',
    lineHeight: 26,
  },
});
