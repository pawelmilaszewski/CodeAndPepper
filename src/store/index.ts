import {createStore} from 'redux';
import reducer from './main/reducer';

export type RootState = ReturnType<typeof reducer>;

export const store = createStore(reducer);
