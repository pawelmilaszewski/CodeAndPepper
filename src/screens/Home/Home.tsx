import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import {useSelector, useDispatch} from 'react-redux';
import {useNavigation} from '@react-navigation/native';
import {StackNavigationProp} from '@react-navigation/stack';

import {ScoreComponent} from '../../components';
import {RootState} from '../../store';
import {resetScores} from '../../store/main/actions';
import {MainParamList} from '../../navigation/Navigator';
import {styles} from './Home.styles';
import {Colors} from '../../utils/colors';

type ScreenProp = StackNavigationProp<MainParamList>;

const Home = () => {
  const navigation = useNavigation<ScreenProp>();
  const dispatch = useDispatch();
  const bestTopScore = useSelector((state: RootState) => state.bestTopScore);
  const bestBottomScore = useSelector(
    (state: RootState) => state.bestBottomScore,
  );

  const reset = () => dispatch(resetScores());

  const goToGame = () => navigation.navigate('Game');

  return (
    <View style={styles.main}>
      <Text style={styles.title}>Star Wars Game</Text>
      <Text style={styles.text}>Your best scores: </Text>
      <ScoreComponent isTop bestScore={bestTopScore} />
      <ScoreComponent bestScore={bestBottomScore} />
      <TouchableOpacity
        onPress={goToGame}
        style={styles.playButton}
        testID={'play_button'}>
        <Text style={styles.playButtonText}>Play!</Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={reset}
        style={[styles.playButton, {backgroundColor: Colors.RED}]}
        testID={'reset_button'}>
        <Text style={styles.playButtonText}>Reset scores</Text>
      </TouchableOpacity>
    </View>
  );
};

export default Home;
