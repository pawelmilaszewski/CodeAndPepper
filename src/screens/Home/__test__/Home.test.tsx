import React from 'react';
import Home from '../Home';
import {render} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import {store} from '../../../store';

const MockHomeComponent = () => (
  <Provider store={store}>
    <Home />
  </Provider>
);

test('render home', () => {
  const tree = render(<MockHomeComponent />);

  const {toJSON, getByText, getByTestId} = tree;

  const title = getByText('Star Wars Game');
  expect(title).toBeTruthy();

  const subtitle = getByText('Your best scores:');
  expect(subtitle).toBeTruthy();

  const button = getByTestId('play_button');
  expect(button).toBeTruthy();

  expect(toJSON()).toMatchSnapshot();
});
