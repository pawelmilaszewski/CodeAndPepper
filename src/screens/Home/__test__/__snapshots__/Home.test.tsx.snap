// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render home 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 20,
      "paddingTop": 54,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32,
        "letterSpacing": 1,
        "lineHeight": 36,
        "textAlign": "center",
      }
    }
  >
    Star Wars Game
  </Text>
  <Text
    style={
      Object {
        "fontSize": 24,
        "letterSpacing": 1,
        "lineHeight": 26,
        "marginBottom": 16,
        "marginTop": 32,
      }
    }
  >
    Your best scores: 
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginVertical": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "lineHeight": 24,
        }
      }
      testID="text"
    >
      Top: 
      0
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 16,
          "opacity": 1,
        }
      }
      testID="reset_button"
    >
      <Text
        style={
          Object {
            "color": "#FF0000",
            "fontSize": 16,
            "lineHeight": 20,
          }
        }
      >
        Reset score
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginVertical": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "lineHeight": 24,
        }
      }
      testID="text"
    >
      Bottom: 
      0
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginLeft": 16,
          "opacity": 1,
        }
      }
      testID="reset_button"
    >
      <Text
        style={
          Object {
            "color": "#FF0000",
            "fontSize": 16,
            "lineHeight": 20,
          }
        }
      >
        Reset score
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#7CFC00",
        "borderRadius": 10,
        "justifyContent": "center",
        "marginTop": 64,
        "opacity": 1,
        "padding": 16,
        "width": "80%",
      }
    }
    testID="play_button"
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "bold",
          "lineHeight": 26,
        }
      }
    >
      Play!
    </Text>
  </View>
</View>
`;
